docs/how-to/websites-on-ipfs/gateway-redirects.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Gateway redirects
+description: What gateway redirects are and how to use them with a website on IPFS.
+---
+
+# THIS IS A DRAFT. DO NOT MERGE.
+
+# To do:
+1. How to redirect old URL to a new place (301 and 302 redirects)
+1. How to use it for PWA/SPA hosting (catch-all 200) 
+1. How to use it to provide custom `404 not found` pages (superceding what `ipfs-404.html` does - ideally not mention old way)
+
+# Gateway Redirects
+
+Gateway Redirects provide support for URL redirects and rewrites for websites hosted on Subdomain or DNSLink gateways. This feature enables support for single-page applications, and avoids link rot when moving to IPFS-backed hosting.
+
+Using Gateway Redirects, you can change the appearance of a URL, change where content is located without breaking existing links, redirect invalid URLs to a custom 404 page, and enable URL rewriting.
+
+# Supported HTTP status codes
+
+* `200` - OK (redirect will be treated as a rewrite, returning OK without changing the URL shown in the browser).
+* `301` - Permanent redirect (the default status).
+* `302` - Found (commonly used for temporary redirects).
+* `303` - See other (replaces PUT and POST with GET).
+* `307` - Temporary redirect (preserves the body and HTTP method of the original request).
+* `308` - Permanent redirect (preserves the body and HTTP method of the original request).
+* `404` - Not found (can be used redirect to custom 404 pages).
+* `410` - Gone (the requested content has been permanently removed).
+* `451` - Unavailable for legal reasons.
+
+# How to set up gateway redirects
+
+To use Gateway Redirects, there must be a file named `_redirects` stored underneath the root CID of the website. This `_redirects` file must be a text file containing one or more lines that follow the format explained below.
+
+## Format of the `_redirects` file
+
+Each line contained within the `_redirects` file has 3 basic components:
+
+1. The `from` path, this specifies the path to be redirected from.
+1. The `to` path, this specifies the path to be redirected to.
+1. The `status` component, this part is optional and specifies the HTTP status code that will be returned. (301, 404, etc.)
+
+For example, if I want to redirect a page to a custom `404` page, the `_redirects` file will contain a line that looks something like this:
+```
+/<requested page> /custom404.html 404
+```
```

Comment:
Probably better to replace this with 302 redirect, and then link to  "Examples" section with things from the TODO at the top of the document:

> **To do**
> 1. How to redirect old URL to a new place (301 and 302 redirects)
> 1. How to use it for PWA/SPA hosting (catch-all 200) 
> 1. How to use it to provide custom `404 not found` pages (superceding what `ipfs-404.html` does - ideally not mention old way)